### PR TITLE
Adding more retriable error types for push retry logic

### PR DESCRIPTION
**Pull request by jhadvig**
QE reported me that scaling the the registry down to zero replicas after the image is successfully pushed, but the process is still going on, the push will fail by one of these errors:
- `Error pushing to registry: Post http://172.30.221.187:5000/v2/test/origin-ruby-sample/blobs/uploads/: net/http: transport closed before response was received`
- `API error (500):  v1 ping attempt failed with error: Get http://172.30.221.187:5000/v1/_ping: read tcp 172.30.221.187:5000: connection reset by peer`

I've added those tested strings and update the logic.

@bparees PTAL

### CODE REVIEW

**Comment by jhadvig** [comment]
[test]

File: pkg/build/builder/dockerutil.go [hunk 1]
```diff
@@ -52,7 +60,12 @@ func pushImage(client DockerClient, name string, authConfig docker.AuthConfigura
 		}
 
 		errMsg := fmt.Sprintf("%s", err)
-		if !strings.Contains(errMsg, "ping attempt failed with error") && !strings.Contains(errMsg, "is already in progress") {
+		for _, errorString := range RetriableErrors {
+			if strings.Contains(errMsg, errorString) {
+				retriableError = true
```
Comment (bparees) [inline pkg/build/builder/dockerutil.go]: you can break out once you find a match.

**Comment by bparees** [comment]
one nit and lgtm.

**Comment by jhadvig** [comment]
@bparees comment addressed :)

**Comment by bparees** [comment]
lgtm [merge]

**Pull request merged**